### PR TITLE
Minor bug fixes

### PR DESCRIPTION
**Pull request by Angelite191**
- tweak: You can now label evidence and sample bags with a pen.
- bugfix: Dialogue for library shelves now have a cancel button.
- bugfix: Kinda fixed Ctrl + Clicking for releasing a pulled object, it behaves slightly differently now. Ctrl + click something unmovable (the floor) to stop pulling the object.

This took extremely long because I'm a scrub, but ey, it's done!

### CODE REVIEW

**Comment by DeityLink** [comment]
> Ctrl + click something unmovable (the floor) to stop pulling the object.
er...what? oh god why. Is there a reason why you can't ctrl+click the object you're pulling?

File: code/modules/library/lib_items.dm [hunk 1]
```diff
@@ -94,6 +95,8 @@
 			else
 				choice.loc = get_turf(src)
 			update_icon()
+		if(choice == "Cancel")
```
Comment (ComicIronic) [inline code/modules/library/lib_items.dm]: This has no code impact.
If you want the user to cancel, there has to be a Cancel button and this has to come before the previous code block.

**Comment by Angelite191** [comment]
I tried a toggle, but it'll just continue pulling the object. I don't know how to check for WHAT is being pulled. Just if the user is pulling.

**Comment by Angelite191** [comment]
I also just realized that you can also stop pulling by clicking something out of range...

**Comment by d3athrow** [comment]
You can already stop pulling something by ctrl+clicking it again, and its mob.pulling and pulling.pulled == mob

**Comment by Angelite191** [comment]
Nah, mang, that's been broken for quite some time now, I've even asked about it in OOC and people said it was borked. If it worked as intended, I wouldn't have bothered with the file.

File: code/modules/library/lib_items.dm (hunk 1 above)
Comment (Angelite191) [inline code/modules/library/lib_items.dm]: I realized this, I just forgot to remove it. Thanks.

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -246,12 +246,15 @@
 	A.CtrlClick(src)
 	return
 /atom/proc/CtrlClick(var/mob/user)
+	if("pull")
+		usr.stop_pulling()
```
Comment (ComicIronic) [inline code/_onclick/click.dm]: user, not usr.
The if also does nothing.
I'd move this under the AM CtrlClick, and use
`else if(user.pulling == src)
   user.stop_pulling()`

**Comment by d3athrow** [comment]
https://github.com/d3athrow/vgstation13/commit/aa2c9bc7dc264741bae58061cae59290021a56ec

**Comment by Angelite191** [comment]
Uuuuh, so can we leave it in anyway? Being able to stop dragging by ctrl clicking is a good thing. And now that we can go back to ctrl clicking the thing we're dragging, it just makes me change a useless fluff.

File: code/modules/detectivework/evidence.dm [hunk 3]
```diff
@@ -79,6 +79,15 @@
 		icon_state = "evidenceobj"
 	return
 
+obj/item/weapon/evidencebag/attackby(obj/item/weapon/W as obj, mob/living/user as mob)
+	if(istype(W, /obj/item/weapon/pen))
+		var/new_label = input("What should the new label be?","")
```
Comment (d3athrow) [inline code/modules/detectivework/evidence.dm]: input needs to be sanitized and the length limited.

Comment (Angelite191) [inline code/modules/detectivework/evidence.dm]: Reference to example please?

Comment (d3athrow) [inline code/modules/detectivework/evidence.dm]: var/new_label = sanitize(trim(input("What should the new label be", "") as null|text))

**Pull request merged**